tests/example_9_nuclear_dynamics/test_ethd_2D.py

```diff
@@ -0,0 +1,428 @@
+import aux_functs
+
+class tmp:
+    pass    
+
+def model1(q, params, full_id):
+    """
+    Symmetric Double Well Potential in 2D
+    Hdia = 0.25*(q1^4 + q2^4) - 0.5*(q1^2 + q2^2)   
+    Sdia = 1.0
+    Ddia = 0.0
+    """
+
+    Id = Cpp2Py(full_id)
+    indx = Id[-1]
+
+    # Hdia and Sdia are ndia x ndia in dimension
+    Hdia = CMATRIX(1,1)
+    Sdia = CMATRIX(1,1)
+
+    # d1ham and dc1_dia are ndia x ndia in dimension, but we have nnucl of them
+    d1ham_dia = CMATRIXList();
+    dc1_dia = CMATRIXList();
+
+    for i in xrange(2):
+        d1ham_dia.append( CMATRIX(1,1) )
+        dc1_dia.append( CMATRIX(1,1) )
+
+    x = q.col(indx).get(0)
+    y = q.col(indx).get(1)
+    x2 = x*x
+    y2 = y*y
+
+    Hdia.set(0,0, (0.25*(x2*x2 + y2*y2) - 0.5*(x2 + y2))*(1.0+0.0j) )
+    Sdia.set(0,0, 1.0+0.0j)
+
+    #  d Hdia / dR_0
+    d1ham_dia[0].set(0,0, x*(x2 - 1.0)*(1.0+0.0j) )
+    d1ham_dia[1].set(0,0, y*(y2 - 1.0)*(1.0+0.0j) )
+
+    #  <dia| d/dR_0| dia >
+    dc1_dia[0].set(0,0, 0.0+0.0j)
+    dc1_dia[1].set(0,0, 0.0+0.0j)
+
+    obj = tmp()
+    obj.ham_dia = Hdia
+    obj.ovlp_dia = Sdia
+    obj.d1ham_dia = d1ham_dia
+    obj.dc1_dia = dc1_dia
+
+    return obj
+
+def model2(q, params, full_id):
+    """
+    The second of the two model potential described by L. Wang, C.C. Martens, 
+    and Y. Zheng, "Entangled trajectory molecular dynamics in 
+    multidimensional systems: Two-dimensional quantum tunneling through 
+    the Eckart barrier" J. Chem. Phys. 137, 34113 (2012)
+
+    Specifically, this potential is Model II 
+
+    Hdia = Va*sech^2*(2q1) + 0.5*Vb*[q2 - Vc(q1^2 - 1)]^2
+
+    Va = barrier height     = 0.00625    
+    Vb = harmonic potential = 0.0106
+    Vc = coupling constant  = 0.4  
+
+    Sdia = 1.0
+    Ddia = 0.0
+    """
+
+    # Define potential specific constants
+    Va = 0.00625    
+    Vb = 0.0106
+    Vc = 0.4    
+
+    Id = Cpp2Py(full_id)
+    indx = Id[-1]
+
+    # Hdia and Sdia are ndia x ndia in dimension
+    Hdia = CMATRIX(1,1)
+    Sdia = CMATRIX(1,1)
+
+    # d1ham and dc1_dia are ndia x ndia in dimension, but we have nnucl of them
+    d1ham_dia = CMATRIXList();
+    dc1_dia = CMATRIXList();
+
+    for i in xrange(2):
+        d1ham_dia.append( CMATRIX(1,1) )
+        dc1_dia.append( CMATRIX(1,1) )
+
+    x = q.col(indx).get(0)
+    y = q.col(indx).get(1)
+    x2 = x*x
+    y2 = y*y
+
+    Hdia.set(0,0,(Va*sech(2.0*x)*sech(2.0*x)+0.5*Vb*(y+Vc*(x*x-1.0))**2)*(1.0+0.0j))
+    Sdia.set(0,0,1.0+0.0j)
+
+    #  d Hdia / dR_0
+    d1ham_dia[0].set(0,0, ( ( -4.0*Va*math.tanh(2.0*x)*sech(2.0*x)**2 ) + ( 2.0*Vb*Vc*x*(y + Vc*x*x - Vc) ) )*(1.0+0.0j))
+    d1ham_dia[1].set(0,0, ( Vb * ( y + Vc*(x*x-1.0) ) ) * (1.0+0.0j))
+
+    #  <dia| d/dR_0| dia >
+    dc1_dia[0].set(0,0, 0.0+0.0j)
+    dc1_dia[1].set(0,0, 0.0+0.0j)
+
+    obj = tmp()
+    obj.ham_dia = Hdia
+    obj.ovlp_dia = Sdia
+    obj.d1ham_dia = d1ham_dia
+    obj.dc1_dia = dc1_dia
+
+    return obj
+
+
+def model3(q, params, full_id):

```

Comment:
This should go into the models_Martens.py module. In both cases, please adhere to the structure used in the models_Libra and models_Tully files - that is don't pass the full_id parameter to this function.